### PR TITLE
Fiat currency updates

### PR DESCRIPTION
**Pull request by danimoh**

### CODE REVIEW

File: src/lib/CurrencyInfo.js [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+    'VES', // sub-unit rarely used
+]);
+
+/**
+ * Cached auto-generated CurrencyInfos such that they do not need to be recalculated.
+ * @private
+ * @readonly
+ * @type {{[currencyAndLocale: string]: CurrencyInfo}}
+ */
+CurrencyInfo.CACHED_AUTO_GENERATED_CURRENCY_INFOS = {};
```
Comment (sisou) [inline src/lib/CurrencyInfo.js:279]: Do we really need to cache currency infos? How long does it actually take to calculate?

File: src/lib/NumberFormatting.js [hunk 2]
```diff
@@ -85,6 +109,11 @@ class NumberFormatting { // eslint-disable-line no-unused-vars
             case 'cny':
                 return 'zh';
             default:
+                // Return the country from the currency code which is typically (but not necessarily) the first two
+                // letters (see https://en.wikipedia.org/wiki/ISO_4217#National_currencies), in the hope that it
+                // coincides with a locale.
+                // TODO oftentimes this results in the wrong locale, e.g. ARS (Argentinan Peso) -> AR (Arabic),
+                //  CAD (Canadian Dollar) -> CA (Catalan). Can we come up with a better heuristic?
```
Comment (sisou) [inline src/lib/NumberFormatting.js:116]: For these cases, they need to be added to the excemption list above, like for EUR and GBP.

**Review by sisou — APPROVED** [review]
I think the tests from @nimiq/utils should also be ported.

**Review by mraveux — APPROVED** [review]
Nice job, well done. And seems to work perfectly 👍 

(i agree with sisou about the tests, if that's not too much effort)

**Review by sisou — APPROVED** [review]
Good to go!

**Pull request merged**